.gitlab/generate-tracer.php

```diff
@@ -496,7 +521,26 @@ function before_script_steps() {
     - if [[ "$MAKE_TARGET" != "test_composer" ]] || ! [[ "$PHP_MAJOR_MINOR" =~ 8.[01] ]]; then sudo composer self-update --$COMPOSER_VERSION --no-interaction; fi
     - COMPOSER_MEMORY_LIMIT=-1 composer update --no-interaction # disable composer memory limit completely
     - make composer_tests_update
-    - for host in ${WAIT_FOR:-}; do wait-for $host --timeout=180; done
+    - |
+      if [ -n "${WAIT_FOR:-}" ]; then
+        for service in $WAIT_FOR; do
+          host=$(echo $service | cut -d: -f1)
+          port=$(echo $service | cut -d: -f2)
+
+          # Detect service type from hostname
+          service_type="generic"
+          case $host in
+            test-agent) service_type="test-agent" ;;
+            mysql-integration) service_type="mysql" ;;
+            elasticsearch*) service_type="elasticsearch" ;;
+            kafka*) service_type="kafka" ;;
+            redis*) service_type="redis" ;;
+            httpbin*) service_type="httpbin" ;;
+          esac
+
+          .gitlab/wait-for-service-ready.sh "$host" "$port" "$service_type" 30 5
+        done
```

Comment:
I am not too deep into the rest of this script, but this is duplicated from above (lines 215 to 234). Would it makes sense to move this into the `.gitlab/wait-for-service-ready.sh`? This way we'd get rid of the duplication and also don't need to keep it in sync in the future.